sympy/sets/sets.py

```diff
@@ -1738,23 +1738,25 @@ def _intersect(self, other):
         return self.__class__(el for el in self if el in other)
 
     def _complement(self, other):
-        if other is S.Reals:
+        if isinstance(other, Interval):
             nums = sorted(m for m in self.args if m.is_number)
-            syms = [m for m in self.args if m.is_Symbol]
-            # Reals cannot contain elements other than numbers and symbols.
+            if other is S.Reals and nums != []:
```

Comment:
@gxyd please also make this change at [L932](https://github.com/sympy/sympy/pull/9682/files?diff=split#diff-7b3e9b0852f1b1d094f87233f56e201fL913)